src/app/quote/quote.py

```diff
@@ -121,8 +121,36 @@ def _build_nvidia_payload(nonce_hex: str, evidences: list) -> str:
     return json.dumps(data)
 
 
+def _derive_key_from_kms(path: str, purpose: str = "signing") -> bytes:
+    """Derive a deterministic key from KMS.
+
+    Args:
+        path: Key derivation path (e.g., "ed25519-signing-key")
+        purpose: Purpose string for key derivation (default: "signing")
+
+    Returns:
+        32-byte key material as bytes
+    """
+    try:
+        client = DstackClient()
+        key_response = client.get_key(path=path, purpose=purpose)
+        key_bytes = key_response.decode_key()
+        if len(key_bytes) != 32:
+            raise ValueError(f"Expected 32-byte key, got {len(key_bytes)} bytes")
+        return key_bytes
+    except Exception as e:
+        log.error(f"Failed to derive key with path '{path}': {type(e).__name__}")
+        raise
+
+
 def _create_ed25519_context() -> SigningContext:
-    private_key = Ed25519PrivateKey.generate()
+    try:
+        key_bytes = _derive_key_from_kms("ed25519-signing-key", purpose="signing")
+        private_key = Ed25519PrivateKey.from_private_bytes(key_bytes)
+    except Exception as e:
+        private_key = Ed25519PrivateKey.generate()
```

Comment:
Should we fail fast if we fail to get the key from KMS? It seems that this feature will be used by a lot of users. Random keys will be an issue in the future